### PR TITLE
Add tslint, prettier and precommit check for lint

### PR DESCRIPTION
**Pull request by nutstick**
# Changes
- Add `tslint`, `prettier`, `husky` and `lint-staged`
- Add precommit hook to trigger `lint-staged` for running lint check before every commit on js/ts files
  - `yarn lint`
  - `prettier`
- Clean up `package.json`
- Add .vscode settings for workspace
  - prettier

# Preview

When try to commit file, pre-commit hook will run on terminal like this, if it fail will stop commit

screenshot 2019-03-04 at 11 04 15 pm

screenshot 2019-03-04 at 11 19 09 pm

### CODE REVIEW

**Comment by nutstick** [comment]
Fix #3

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+language: node_js
```
Comment (nutstick) [inline .travis.yml]: Make another PR please

**Review by Panupong-Tongtawach — APPROVED** [review]
